### DISCUSSION TITLE
1주차 미팅

### DISCUSSION THREAD

**Discussion by nyj001012** [GunGonGamLee/ft_transcendence#12] [General]
# 의논할 것
- API 설계
- 기능 구현 순서
- 게임 모드 방식

**Comment by sejoonkimmm** [comment]
* About 게임 모드
1. 토너먼트 방식 (자동 큐 매칭 시스템, 방 없음)
-> 4명이서 큐를 돌려 매칭이 되면 바로 시작된다. 즉, 이긴 플레이어만 2게임을 하게 된다.
-> 1등은 Rating +10점, 2등은 +5점 먼저 진 놈은 -5점을 때리는 상남자의 게임

2. 사용자 지정 게임 (방 만들기)
-> 1vs1 게임만 가능하다, 이걸로 토너먼트를 할 수는 없음!
-> 승리, 패배 시 전적은 기록되나(n승 n패) Rating에는 반영하지 않는다.

3. 2P 모드
-> 승리, 패배 시 승리, 패배 자체는 기록되나 n승 n패에 기록되지 않고, Rating에도 반영되지 않는다.
-> 그냥 시작만 되면 된다.

**Reply by nyj001012** [reply]
1. 토너먼트 방식
   1.1. 4명 매칭이 되면, 프론트에 4명의 nickname을 반환값으로 전달

**Comment by sejoonkimmm** [comment]
1월 3일 

TODO LIST

- [x] 친구 추가 및 삭제, 데이터 베이스 설계
- [x] 무시하기(차단) 데이터 베이스 설계

**Comment by LWJ0513** [comment]
![이상해씨](http://file3.instiz.net/data/file3/2018/01/31/e/d/3/ed3813208af4e2ccf1cd009db099db1f.gif)

## 만들어야 하는 페이지

### 로그인 페이지
- 42 로그인
- 구글 로그인

### 2단계 인증(2FA) 페이지
- 2FA
- 처음 로그인 하는 사람이면 modal + ajax(사용자 닉네임 검증용) 로 닉네임 입력 받기

### 메인 페이지
- 사용자 닉네임
- 친구 목록
   - DM
- 채팅
   - 전체 채팅
   - 토너먼트 시작 전 알림
- 게임 시작 버튼 (게임 모드 선택 페이지로 이동)

### 게임 모드 선택 페이지
- 토너먼트 방식
- 사용자 지정 게임 (게임 방 만들기 페이지)
   - 클릭시 방 이름 받는 모달
- 2P 모드

 ### 게임 페이지
- 모드에 따라 UI를 다르게 출력한다.
- 게임 상대방 닉네임
- 게임 화면
  
### 사용자 정보 페이지
- 본인의 페이지일 경우 '로그아웃', 다른 사람의 페이지일 경우 '친구 추가'
- 사용자 게임 점수(포인트)
- 사용자 전적



## TODO
- [ ] 차단한 친구 목록
- [ ] 게임 대기 화면 ( 토너먼트, 1대1)
- [ ] 채팅방

**Comment by sejoonkimmm** [comment]
1월 4일
- [x] 모노레포 변경하기 (프로젝트 구조 바꾸기)
- [x] nginx container 띄우고 443 포트 열어두기
- [x] django + node.js 컨테이너 작성하기
